onlyoffice-nextcloud/docker-compose.yml

```diff
@@ -0,0 +1,33 @@
+version: '3.9'
+services:
+  app_proxy:
+    environment:
+      APP_HOST: onlyoffice-nextcloud_web_1
+      APP_PORT: 3000
+
+  web:
+    image: a4004/onlyoffice-nextcloud-web:1.0.4@sha256:774bd9bda641e6be49286921637840efba050fdfd022934dfd9ee9b6ec8d0a63
+    restart: on-failure
+    environment:
+      - DOCS_ADDRESS=http://$DEFAULT_INTERFACE_IP:$DOCSERVER_PORT
+      - DOCS_INTERNAL_ADDRESS=http://$DEFAULT_INTERFACE_IP:$DOCSERVER_PORT
+      - NEXTCLOUD_INTERNAL_ADDRESS=http://$DOCKER_INTERFACE_IP:$APP_NEXTCLOUD_PORT
```

Comment:
I think we can simplify this a bit and avoid needing to run ip-related commands in the `exports.sh`. I'm not very familiar with the Nextcloud/ONLYOFFICE ecosystem, so please let me know if anything I suggest here doesn't make sense or has unintended consequences:

1) `DOCS_ADDRESS=http://$DEFAULT_INTERFACE_IP:$DOCSERVER_PORT`
I'm thinking that most users will be accessing Nextcloud via `umbrel.local` not their IP address, so if they are using their local IP address in the nextcloud/onlyoffice config they will get hit with cors errors when clicking on a document in Nextcloud from `http://umbrel.local:8081` and it tries to make a request to `$DEFAULT_INTERFACE_IP:5672`. 

Shall we change this to `DOCS_ADDRESS=http://$DEVICE_DOMAIN_NAME:$DOCSERVER_PORT` (which will be http://umbrel.local:5672) and then include instructions saying that if a user is accessing Nextcloud from a different hostname/ip, they should change make the appropriate change to `ONLYOFFICE Docs address` in their Nextcloud settings?

2) `DOCS_INTERNAL_ADDRESS=http://$DEFAULT_INTERFACE_IP:$DOCSERVER_PORT`
This is for internal requests to the `documentserver` container, so we can just use the container name here and Docker's internal DNS will resolve it to the correct IP address. So I think we can change this to `DOCS_INTERNAL_ADDRESS=http://onlyoffice-nextcloud_documentserver_1:$DOCSERVER_PORT`.
The user will then copy-paste `http://onlyoffice-nextcloud_documentserver_1` into the ONLYOFFICE settings in Nextcloud.

3) `NEXTCLOUD_INTERNAL_ADDRESS=http://$DOCKER_INTERFACE_IP:$APP_NEXTCLOUD_PORT`
I think we can get away without the export.sh ip command here as well. We actually have access to the gateway IP for the umbrel docker network as an env var. Check it out: https://github.com/getumbrel/umbrel/blob/570acdabcab13ac237fba992f491d126fff30810/packages/umbreld/source/modules/apps/legacy-compat/app-script#L10
So we should be able to use `NEXTCLOUD_INTERNAL_ADDRESS=http://$GATEWAY_IP:$APP_NEXTCLOUD_PORT` which will yield `http://10.21.0.1:8081/` for the user to copy-paste into the ONLYOFFICE settings in Nextcloud.

So a user would end up with this:

<img width="474" alt="image" src="https://github.com/user-attachments/assets/d68fe7ae-e7a2-499a-8a4f-a6ffe8018be5">

Reply:
@nmfretz  Is it not better to use something like `http://nextcloud_web_1:8081`? We are using container hostname reference on torrenting apps like Prowlarr, Radarr and Sonarr.

![image](https://github.com/user-attachments/assets/a4ed6cde-f6ef-4841-926d-36d1000f6500)

Reply:
That's actually a good point. 

I have tested both `http://nextcloud_web_1/` and `http://nextcloud_web_1:8081/` but it doesn't appear to work for some reason. Nextcloud is throwing an error `Error when trying to connect (Error occurred in the document service: Error while downloading the document file to be converted.) (version 8.1.1.26)`.

Perhaps I'm doing something wrong here or it's not set up for this type of communication, not too sure but I agree with the idea of using container hostnames over IPs for sure.

Reply:
@joaovictor-local @a4004, ya the issue here is that the Nextcloud app doesn't have the `web` container name as a trusted proxy in its `config.php`, so instead we are using an IP that it does trust (e.g., `10.21.0.1` or `172.17.0.1`).

https://github.com/getumbrel/umbrel-apps/blob/49580db282c58ac50f077f2d83f998f71f460d17/nextcloud/docker-compose.yml#L47
https://github.com/getumbrel/umbrel-apps/blob/49580db282c58ac50f077f2d83f998f71f460d17/nextcloud/exports.sh#L3-L4

For this to work, the Nextcloud config.php would have to include `nextcloud_web_1/` under trusted_domains. Then in the field for `Server address for internal requests from ONLYOFFICE Docs` a user would use `http://nextcloud_web_1/`, without a port specified since we want the default port 80 **inside the web container** (not 8081).

I do really like the simplicity of using container names @joaovictor-local, so we could consider the following:

- Update the Nextcloud app to include `nextcloud_web_1/` in the config.php
- Change the env var in ONLYOFFICE Docs to `NEXTCLOUD_INTERNAL_ADDRESS=http://nextcloud_web_1`
- Include a note in the `a4004/onlyoffice-nextcloud-web` UI (or maybe just the app description) telling users to update Nextcloud if they haven't.

What do you guys think?

Reply:
Update: I have updated Nextcloud to allow connection via nextcloud_web_1 https://github.com/getumbrel/umbrel-apps/pull/1297